frontend-web/src/shared/persistence.ts

```diff
@@ -0,0 +1,18 @@
+const keys = {
+  username: 'username'
+} as const
+
+type ActualKeys = typeof keys[keyof typeof keys];
+const inMemoryCache: { [key in ActualKeys]?: string } = {};
+
+export const storeUsername = (value: string) => storeString(keys.username, value);
+export const getUsername = () => getString(keys.username);
+
+function storeString(key: ActualKeys, value: string) {
```

Comment:
Minor: Normally I would try to avoid types in method names because I can see the type on parameter and you need to remember to rename the method once you change the parameter type

Reply:
agreed, but I put it there to make it clear that this function can't handle numbers. If we ever need to store numbers, too, we have to have (at least) a `loadNumber` function, because even if we had a `load<T>(key:ActualKeys)` the type parameter `T` wouldn't influence the runtime behavior.

Reply:
Maybe you could avoid this problem if you json serialize the value before save and deserialize on load

Reply:
Yes, we could combine serialization with a template parameter :thinking:

Reply:
I have tried multiple things now, but no combination of ideas really improved this. I'd say we cross that bridge when we come to it.

Reply:
I've changed it after all, because I realized that if we don't at least store this properly now, then we can't go back without migrations and stuff. How do you like https://github.com/erlangen-dev/coffee-caller/compare/b9891292a1c3d108a4997d5b3b78229bd93002f7..78ef9af2ea8722bf2a459f050c96ddd0b1237bd9

Reply:
<3